torch/csrc/jit/script/lexer.h

```diff
@@ -190,6 +191,46 @@ struct SharedParserData {
     *len = endptr - startptr;
     return *len > 0;
   }
+
+  bool isCharCount(char c, const std::string& str, size_t start, int len) {
+    //count checks from [start, start + len) 
+    return start + len <= str.size() && std::count(str.begin() + start, str.begin() + start + len, c) == len; 
+  }
+
+  // python conconcatenates all adjacent strings "a" "b" == "ab"
+  // strings can be enclosed with 1 or 3 single or double quotes
+  // if enclosed 3 quotes newlines are valid
+  // as elsewhere, backslash and new line should be ignored
+  bool isString(const std::string& str, size_t start, size_t* len) {
+    char quote = str[start];
+    if (quote != '\"' && quote != '\'')
+      return false;
+    int quote_len = isCharCount(quote, str, start, 3) ? 3 : 1; 
+    size_t end = start + quote_len; 
+    while(end < str.size() && !isCharCount(quote, str, end, quote_len)) {
+      if (str[end] == '\n' && quote_len != 3) {
+        return false;
+      }
+      if (str[end] == '\\') {
```

Comment:
What's this rule for?

Reply:
This seems like it could make `end  == str.size() + 1` if the parsed texted ended in a backslash, and the fail the `end == str.size()` check below.

Reply:
This ignores escaped quotations or escaped new lines,  ^ as your above question asks.